android/src/main/java/com/pspdfkit/flutter/pspdfkit/PspdfkitPlugin.java

```diff
@@ -90,4 +163,34 @@ private boolean isImageDocument(@NonNull String documentPath) {
         }
         return extension.equals("png") || extension.equals("jpg") || extension.equals("jpeg");
     }
+
+    @Override
+    public boolean onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        if (permissions.length == 0) return false;
+
+        int status = 0;
+        String permission = permissions[0];
+        if (requestCode == 0 && grantResults.length > 0) {
+            if (ActivityCompat.shouldShowRequestPermissionRationale(registrar.activity(), permission)) {
+                // Permission denied.
+                status = 1;
+            } else {
+                if (ActivityCompat.checkSelfPermission(registrar.context(), permission) == PackageManager.PERMISSION_GRANTED) {
+                    // Permission allowed.
+                    status = 2;
+                } else {
+                    // Set to never ask again.
+                    Log.i(LOG_TAG, "Set to never ask again " + permission);
+                    status = 3;
+                }
+            }
+        }
+        Log.i(LOG_TAG, "Requesting permission status: " + status);
+
+        Result result = permissionRequestResult.getAndSet(null);
```

Comment:
Using this implementation guarantees that the result is correctly nullified, and I was not able to observe the issue anymore